src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/Completion/CompletionHandler.cs

```diff
@@ -99,10 +109,29 @@ private static CompletionItem CreateCompletionItem(XamlCompletionItem xamlComple
                     Title = CreateEventHandlerCommandTitle
                 };
             }
+            else if (xamlCompletion.RetriggerCompletion)
+            {
+                // Retriger completion after commit
+                item.Command = new Command()
+                {
+                    CommandIdentifier = StringConstants.RetriggerCompletionCommand,
+                    Title = RetriggerCompletionCommandTitle
+                };
+            }
 
             return item;
         }
 
+        private static SumType<string[], CommitCharacter[]> GetCommitCharacters(XamlCompletionItem completionItem)
+        {
+            if(completionItem.XamlCommitCharacters == null)
+            {
+                return completionItem.CommitCharacters;
+            }
+
```

Comment:
Is this needed since nullable is enabled by default in Roslyn and completionItem is not a nullable?

Reply:
This is actually needed for not breaking our current completion as we haven't made the change in XLS. We have some custom completion types with custom commit characters, for example, namespace values that will need to be passed to the CompletionItem.

Reply:
Yeah, good point.

---
In reply to: [618737717](https://github.com/dotnet/roslyn/pull/52848#discussion_r618737717) [](ancestors = 618737717)